### PR TITLE
fix: check existence of tenant before updating

### LINKED ISSUE #8: ControlPlane-TenantManagement-TenantDelete: Accepts tenant ids that don't exist
### Describe the bug

When running DELETE /tenants/{tenantId}, if tenantId doesn't exist, the service accepts the request, and writes a new record to the TenantDetails DynamoDB table.

> 

### Expected Behavior

When TenantID does not exist, the service should return 404.

### Current Behavior

Returns 200 and writes new entry to TenantDetails table

### Reproduction Steps

invoke DELETE /tenant/tenantId with tenantId that does not exist in the database

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.131.0 (build 92b912d)

### Framework Version

_No response_

### Node.js Version

v18.19.0

### OS

Debian GNU/Linux 12 (bookworm)

### Language

TypeScript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by suhussai**
### Issue # (if applicable)

Closes #8 

### Description of changes

Check if tenant id exists before updating the tenant.

### Checklist

- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/awslabs/sbt-aws/blob/main/CONTRIBUTING.md)

---

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.*

### CODE REVIEW

File: resources/functions/tenant-management/tenant_management.py
Comment (tobuck-aws) [inline resources/functions/tenant-management/tenant_management.py:1]: Are there any tests we can write for this?

Comment (suhussai) [reply resources/functions/tenant-management/tenant_management.py:1]: I've got something cooking, but not ready yet. I am planning on integrating it once it's done with some CICD pipeline that we can hook into GitHub. Not there yet, though.

Comment (suhussai) [reply resources/functions/tenant-management/tenant_management.py:1]: The testing pipeline I was referring to has been merged as part #15 

I've added to that list of tests to include a test for non-existent tenants.

**Review by tobuck-aws — APPROVED** [review]
that works

**Pull request merged**